src/order/closure.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Bhavik Mehta
+-/
+import order.basic
+import order.preorder_hom
+import order.galois_connection
+import tactic.monotonicity
+
+/-!
+# Closure operators on a partial order
+
+We define (bundled) closure operators on a partial order as an monotone (increasing), extensive
+(inflationary) and idempotent function.
+We define closed elements for the operator as elements which are fixed by it.
+
+## References
+
+* https://en.wikipedia.org/wiki/Closure_operator#Closure_operators_on_partially_ordered_sets
+
+-/
+universe u
+
+variables (α : Type u) [partial_order α]
+
+/--
+A closure operator on the partial order `α` is a monotone function which is extensive (every x
+is less than its closure) and idempotent.
+-/
+@[ext]
```

Comment:
What lemma does `@[ext]` generate here (brefore we have `has_coe_to_fun`)?

Reply:
It makes 
```
closure_operator.ext : ∀ (x y : closure_operator α), x.to_preorder_hom = y.to_preorder_hom → x = y
```
which in hindsight isn't the most helpful. I'll change.

Reply:
Amended, thanks.